src/psyclone/domain/lfric/arg_ordering.py

```diff
@@ -0,0 +1,714 @@
+        var_accesses.
+
+        :param arg: the meta-data description of the kernel \
+            argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: because this is an abstract method
+
+        '''
+        raise NotImplementedError(
+            "Error: ArgOrdering.stencil() must be implemented by subclass")
+
+    def operator(self, arg, var_accesses=None):
+        '''Add the operator arguments to the argument list. If supplied it
+        also stores this access in var_accesses.
+
+        :param arg: the meta-data description of the operator.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises NotImplementedError: because this is an abstract method
+
+        '''
+        raise NotImplementedError(
+            "Error: ArgOrdering.operator() must be implemented by subclass")
+
+    def scalar(self, scalar_arg, var_accesses=None):
+        '''Add the name associated with the scalar argument to the argument
+        list and optionally add this scalar to the variable access
+        information.
+
+        :param scalar_arg: the kernel argument.
+        :type scalar_arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance that \
+            stores information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises InternalError: if the argument is not a recognised scalar type.
+
+        '''
+        if not scalar_arg.is_scalar():
+            raise InternalError(
+                "Expected argument type to be one of {0} but got '{1}'".
+                format(LFRicArgDescriptor.VALID_SCALAR_NAMES, scalar_arg.type))
+
+        self.append(scalar_arg.name)
+        if var_accesses is not None:
+            var_accesses.add_access(scalar_arg.name, AccessType.READ,
+                                    self._kern)
+
+    def fs_common(self, function_space, var_accesses=None):
+        '''Add function-space related arguments common to LMA operators and
+        fields. If supplied it also stores this access in var_accesses.
+
+        :param function_space: the function space for which the related \
+            arguments common to LMA operators and fields are added.
+        :type function_space: :py:class:`psyclone.dynamo0p3.FunctionSpace`
```

Comment:
`psyclone.domain.lfric.FunctionSpace`

Reply:
Fixed, in all instance (and I also found a few in dynamo0p3 that I've missed, I've fixed them all).